bentoml/_internal/bento/build_config.py

```diff
@@ -64,9 +64,7 @@ def _convert_python_version(py_version: str | None) -> str | None:
     target_python_version = f"{major}.{minor}"
     if target_python_version != py_version:
         logger.warning(
-            "BentoML will install the latest python%s instead of the specified version %s. To use the exact python version, use a custom docker base image. See https://docs.bentoml.org/en/latest/concepts/bento.html#custom-base-image-advanced",
-            target_python_version,
-            py_version,
+            f"BentoML will install the latest 'python{target_python_version}' instead of the specified 'python{py_version}'. To use the exact python version, use a custom docker base image. See https://docs.bentoml.org/en/latest/concepts/bento.html#custom-base-image-advanced",
```

Comment:
f string will be slower than the original code

Reply:
really?

Reply:
https://docs.python.org/3/howto/logging.html#optimization

Reply:
yup